rust/src/nasl/utils/context.rs

```diff
@@ -329,6 +333,59 @@ impl NaslContext {
     }
 }
 
+#[derive(Debug)]
+pub struct Target {
+    /// The original target. IP or hostname
+    target: String,
+    /// The IP address. Always has a valid IP. It defaults to 127.0.0.1 if not possible to resolve target.
+    ip_addr: IpAddr,
+    // The shared state is guarded by a mutex. This is a `std::sync::Mutex` and
+    // not a Tokio mutex. This is because there are no asynchronous operations
+    // being performed while holding the mutex. Additionally, the critical
+    // sections are very small.
+    //
+    // A Tokio mutex is mostly intended to be used when locks need to be held
+    // across `.await` yield points. All other cases are **usually** best
+    // served by a std mutex. If the critical section does not include any
+    // async operations but is long (CPU intensive or performing blocking
+    // operations), then the entire operation, including waiting for the mutex,
+    // is considered a "blocking" operation and `tokio::task::spawn_blocking`
+    // should be used.
+    /// vhost list which resolve to the IP address and their sources.
+    vhosts: Mutex<Vec<(String, String)>>,
```

Comment:
Comment: I know we talked about this and didnt find a short-term solution that was any better than this mutex, but I think it is a shame that we have to resort to mutexes all over the place instead of finding a more robust solution that passes the responsibility of handling shared/mutable state to the interpreter/executor.